### PR TITLE
minor fixes around modules

### PR DESCRIPTION
**Pull request by 0xEgao**
# Pull Request

## Description
Fixes small bugs -:
1-) removed `timeout` from `net_wait_for_confirmation` api. as generally most of the blocks are not mined under 10min, so wait until the txn is confirmed. 
2-) Currently while testing on signet i was getting a lot of WouldBlock spam from nostr relay response, i.e.
```rust
2026-05-04T00:04:48.302868+05:30 WARN coinswap::watch_tower::nostr_discovery - Nostr session error on wss://relay.damus.io: WebSocket(Io(Os { code: 35, kind: WouldBlock, message: "Resource temporarily unavailable" })), retrying in 5s
2026-05-04T00:04:52.673107+05:30 WARN coinswap::watch_tower::nostr_discovery - Nostr session error on wss://nos.lol: WebSocket(Io(Os { code: 35, kind: WouldBlock, message: "Resource temporarily unavailable" })), retrying in 5s
```
Now, In `read_event_loop`, WouldBlock / TimedOut / Interrupted are now treated as non-fatal (continue loop), to avoid reconnections.

3-) Currently in a typical swap, we have a lot of calls to `zmq::poll` && `zmq::recv_event`, so I added a small idle backoff in the watcher loop.
Changes in Profiling-:
```rust
Earlier
coinswap::watch_tower::zmq_backend::poll                                         114548537    1218.59 s     10.64 µs     16.05 µs    97.12%
coinswap::watch_tower::zmq_backend::recv_event                                   114548537    1168.99 s     10.21 µs     15.46 µs    93.17%

Now
Current
coinswap::watch_tower::zmq_backend::poll                                           67661       2.51 s     37.10 µs     62.11 µs     0.60%
coinswap::watch_tower::zmq_backend::recv_event                                     67662       2.14 s     31.60 µs     45.57 µs     0.51%
```
## Related Issue(s)
Closes #

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code refactor / performance improvement
- [ ] Documentation update only
- [ ] CI / Docker / Build changes
- [ ] Other (please describe):

## Protocol Version(s) Affected
- [ ] Legacy (ECDSA — `messages.rs`, `contract.rs`, `handlers.rs`)
- [ ] Taproot-Musig2 (`messages2.rs`, `contract2.rs`, `handlers2.rs`)
- [ ] Both
- [x] Neither (infrastructure / tooling only)

> **Note:** When modifying protocol logic, changes must be made to **both** versions unless the
> scope is explicitly version-specific.

## Affected Component(s)
- [ ] **makerd** (background daemon)
- [x] **taker** (CLI client)
- [ ] **maker-cli** (command-line tool)
- [ ] Core protocol / cryptography / library
- [x] **watch_tower** (contract monitoring & recovery)
- [ ] Docker / deployment scripts
- [ ] Tests / test framework
- [ ] Documentation (`docs/`)
- [ ] Other (please specify):

### Code Quality
- [x] I ran `cargo +nightly fmt --all` and committed the result
- [x] I ran `cargo +stable clippy --all-features --lib --bins --tests -- -D warnings` with zero warnings
- [x] I ran `cargo +stable clippy --examples -- -D warnings` with zero warnings
- [x] I ran `RUSTDOCFLAGS="-D warnings" cargo +nightly doc --all-features --document-private-items --no-deps` with zero warnings
- [x] Pre-commit git hook passes (`ln -s ../../git_hooks/pre-commit .git/hooks/pre-commit` if not already set)

### Testing
- [ ] All unit tests pass (`cargo test`)
- [ ] Integration tests pass (`cargo test --features integration-test`)
- [ ] Changes were manually tested on **regtest**
- [ ] End-to-end maker ↔ taker swap tested (where applicable)
- [ ] Test-only code is gated behind `#[cfg(feature = "integration-test")]`

### Documentation
- [ ] Relevant files in the `docs/` folder were updated
- [ ] Complex logic is commented

### Security & Privacy (Critical)
- [ ] This change preserves **trustlessness** and **atomic swap guarantees**
- [ ] No regression in **sybil resistance** or **fidelity bond** logic
- [ ] Tor anonymity and P2P message flow were reviewed
- [ ] No new attack vectors or trust assumptions introduced
- [ ] Edge cases and error handling considered
- [ ] ZMQ subscription integrity (raw tx/block feed) is unaffected
- [ ] `integration-test` feature flag is not reachable in production code paths

## Summary by CodeRabbit

* **Bug Fixes**
  * Longer non-test transaction confirmation timeout to reduce false timeouts.
  * Treat transient WebSocket I/O errors as recoverable to keep discovery/monitoring running.
  * Add brief idle sleep in the watcher loop to lower CPU usage when idle.

* **Reliability / UX**
  * Confirmation waiting now uses a wallet-backed, multi-transaction waiter with interrupt/abort support and clearer interruption reporting.

* **Tests**
  * Integration tests updated to match the new confirmation-wait behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/citadel-tech/coinswap/pull/863)

### CODE REVIEW

**Review by mojoX911 — APPROVED** [review]
Ack

File: src/wallet/api.rs [hunk 1]
```diff
@@ -2623,43 +2624,94 @@ impl Wallet {
...
+            // Sleep in 1-second increments so we can check shutdown/abort.
+            for _ in 0..total_sleep {
+                if shutdown.is_some_and(|s| s.load(std::sync::atomic::Ordering::Relaxed)) {
+                    return Err(WalletError::Interrupted("Shutdown requested"));
+                }
+                if abort_check.is_some_and(|f| f()) {
+                    return Err(WalletError::Interrupted("Abort requested"));
+                }
+                thread::sleep(Duration::from_secs(1));
+            }
```
[the remark this answers is not in the document]
Comment (mojoX911) [reply src/wallet/api.rs]: This is good point. We should cap the wait time at 10 mins max. Or else it will cause runaway waits if block is too slow.

**Comment by mojoX911** [comment]
Looks good to me. Feel free to take any of the other feedbacks given by coderabbit.

File: src/taker/taproot_swap.rs [hunk 2]
```diff
@@ -563,7 +563,11 @@ impl Taker {
             .iter()
             .map(|sc| sc.contract_tx.compute_txid())
             .collect();
-        self.net_wait_for_confirmation(&contract_txids, None)?;
+        let required_confirms = self.swap_state()?.params.required_confirms;
+        {
+            let wallet = self.read_wallet()?;
+            wallet.wait_for_tx_confirmation(&contract_txids, required_confirms, None, None)?;
+        }
```
[the remark this answers is not in the document]
Comment (0xEgao) [reply src/taker/taproot_swap.rs:575]: This call passes `abort_check` parameter as none so we cannot receive `WalletError::Interrupted(_)` here, therefore match arms are not required

File: src/taker/legacy_swap.rs [hunk 3]
```diff
@@ -310,7 +310,16 @@ impl Taker {
...
+                let required_confirms = self.swap_state()?.params.required_confirms;
+                prev_confirm_height = {
+                    let wallet = self.read_wallet()?;
+                    wallet.wait_for_tx_confirmation(
+                        &funding_txids,
+                        required_confirms,
+                        None,
+                        None,
+                    )?
+                };
```
[the remark this answers is not in the document]
Comment (0xEgao) [reply src/taker/legacy_swap.rs:323]: Same here match arms are not required as the `abort_check` parameter is none.

File: src/wallet/api.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (0xEgao) [reply src/wallet/api.rs]: Makes sense, added a 10min cap
